.github/workflows/wheels.yml

```diff
@@ -75,6 +80,102 @@ jobs:
+          - arch: x86
+            cibw_arch: x86
+          - arch: x64
+            cibw_arch: AMD64
+          - arch: ARM64
+            cibw_arch: ARM64
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Checkout extra test images
+        uses: actions/checkout@v4
+        with:
+          repository: python-pillow/test-images
+          path: Tests\test-images
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.x"
+
+      - name: Prepare for build
+        run: |
+          choco install nasm --no-progress
+          echo "C:\Program Files\NASM" >> $env:GITHUB_PATH
+
+          # Install extra test images
+          xcopy /S /Y Tests\test-images\* Tests\images
+
+          & python.exe -m pip install -r .ci/requirements-cibw.txt
+
+          # Cannot cross-compile FriBiDi (only used for tests)
+          $FLAGS = ("--no-imagequant", "--architecture=${{ matrix.arch }}")
+          if ('${{ matrix.arch }}' -eq 'ARM64') { $FLAGS += "--no-fribidi" }
+          & python.exe winbuild\build_prepare.py -v @FLAGS
+        shell: pwsh
+
+      - name: Build wheels
+        run: |
+          setlocal EnableDelayedExpansion
+          for %%f in (winbuild\build\license\*) do (
+            set x=%%~nf
+            rem Skip FriBiDi license, it is not included in the wheel.
+            set fribidi=!x:~0,7!
+            if NOT !fribidi!==fribidi (
+              rem Skip imagequant license, it is not included in the wheel.
+              set libimagequant=!x:~0,13!
+              if NOT !libimagequant!==libimagequant (
+                echo. >> LICENSE
+                echo ===== %%~nf ===== >> LICENSE
+                echo. >> LICENSE
+                type %%f >> LICENSE
+              )
+            )
+          )
+          call winbuild\\build\\build_env.cmd
+          %pythonLocation%\python.exe -m cibuildwheel . --output-dir wheelhouse
+        env:
+          CIBW_ARCHS: ${{ matrix.cibw_arch }}
+          CIBW_BEFORE_ALL: "{package}\\winbuild\\build\\build_dep_all.cmd"
+          CIBW_CACHE_PATH: "C:\\cibw"
+          CIBW_TEST_SKIP: "*-win_arm64"
+          CIBW_TEST_COMMAND: 'docker run --rm
+            -v {project}:C:\pillow
+            -v C:\cibw:C:\cibw
+            -v %CD%\..\venv-test:%CD%\..\venv-test
+            -e CI -e GITHUB_ACTIONS
+            mcr.microsoft.com/windows/servercore:ltsc2022
+            powershell C:\pillow\.github\workflows\wheels-test.ps1 %CD%\..\venv-test'
+        shell: cmd
+
+      - name: Upload wheels
+        uses: actions/upload-artifact@v3
+        with:
+          name: dist
+          path: ./wheelhouse/*.whl
+
+      - name: Prepare to upload FriBiDi
+        if: "matrix.arch != 'ARM64'"
+        run: |
+          mkdir fribidi\${{ matrix.arch }}
+          copy winbuild\build\bin\fribidi* fribidi\${{ matrix.arch }}
+        shell: cmd
+
+      - name: Upload fribidi.dll
+        if: "matrix.arch != 'ARM64'"
+        uses: actions/upload-artifact@v3
```

Comment:
Previously, this was only run for one Python version.

https://github.com/python-pillow/Pillow/blob/63bec07b971cfe54311a2883206c020756081fa5/.github/workflows/test-windows.yml#L246-L248

Any particular reason it's now being run for every version?

Reply:
This is now in the `wheels.yml` workflow which only runs a single job for all Python versions of a single architecture.
Previously it was in `test-windows.yml` which has a separate job for each Python version.

Reply:
Yeah, the others were replaced with `pytest.importorskip` (https://github.com/python-pillow/Pillow/pull/4436) which doesn't fit here.